src/HistoryDialog.vala

```diff
@@ -97,9 +98,13 @@ namespace PantheonCalculator {
 
             get_content_area ().add (main_grid);
 
-            add_button (_("Close"), Gtk.ResponseType.CLOSE);
+            // Use progressive custom response code with 0 as the default close action
+            var button_clear = add_button (_("Clear History"), -1);
+            button_clear.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
 
-            var button_add = add_button (_("Insert"), Gtk.ResponseType.OK);
+            add_button (_("Close"), 0);
```

Comment:
It would be better to continue to use `Gtk.ResponseType.CLOSE` here so that the response switch statement below is more easily legible